set.mm

```diff
@@ -725218,6 +725219,65 @@ valid cases ( ` (/) ` is the last symbol) and invalid cases ( ` (/) `
       CCABACBAHIUDUECABACJKUFBADSACBHBALMNOSABCQR $.
   $}
 
+  ${
+    $d b x $.
+    $( Lemma for ~ ichnf1lem3 :  A substitution of a non-free variable has no
+       effect.  Give the distinctor in a form that can be easily eliminiated.
+       (Contributed by Wolf Lammen, 6-Aug-2023.) $)
+    ichnf1lem1 $p |- ( ( A. y F/ x ph /\ -. A. x x = y )
+                    -> ( [ a / x ] [ b / y ] ph <-> [ b / y ] ph ) ) $=
+      ( wnf wal weq wn wa wsb wb naev nfsb4t syl5 imp sbft syl ) ABFCGZBCHBGIZJ
+      ACEKZBFZUABDKUALSTUBTBEHBGISUBBCEBMACEBNOPUABDQR $.
+
+    $( Lemma for ~ ichnf1lem3 :  When substituting successively for two always
+       equal variables, the second substitution has no effect.  (Contributed by
+       Wolf Lammen, 6-Aug-2023.) $)
+    ichnf1lem2 $p |- ( A. x x = y
+                    -> ( [ a / x ] [ b / y ] ph <-> [ b / y ] ph ) ) $=
+      ( weq wal wsb wnf wb nfs1v nfa1 drsb1 nfbidf mpbii sbft syl ) BCFZBGZACEH
+      ZBIZTBDHTJSABEHZBIUAABEKSUBTBRBLABCEMNOTBDPQ $.
+
+    $( Lemma for ~ ichnf1 :  A substitution of a non-free variable has no
+       effect.  (Contributed by Wolf Lammen, 6-Aug-2023.) $)
+    ichnf1lem3 $p |- ( A. y F/ x ph
+                    -> ( [ a / x ] [ b / y ] ph <-> [ b / y ] ph ) ) $=
+      ( wnf wal weq wsb wb ichnf1lem2 adantl ichnf1lem1 pm2.61dan ) ABFCGZBCHBG
+      ZACEIZBDIQJZPROABCDEKLABCDEMN $.
+  $}
+
+  ${
+    $d a b x y $.  $d a b ph $.
+    $( If in an interchangeability context ` x ` is not free in ` ph ` , the
+       same holds for ` y ` .  (Contributed by Wolf Lammen, 6-Aug-2023.)
+       (Revised by AV, 23-Sep-2023.) $)
+    ichnf1 $p |- ( ( A. y F/ x ph /\ [ x <> y ] ph ) -> A. x F/ y ph ) $=
+      ( va vb wnf wal wich nfnf1 nfal nfich1 nfan wsb dfich2 ichnf1lem3 bibi12d
+      wa wb bicom1 syl6bi 2alimdv syl5bi imp sbnf2 sylibr alrimi ) ABFZCGZABCHZ
+      QZACFZBUHUIBUGBCABIJABCKLUJACDMZACEMZRZEGDGZUKUHUIUOUIUMBDMZULBEMZRZEGDGU
+      HUOABCDENUHURUNDEUHURUMULRUNUHUPUMUQULABCDEOABCEDOPUMULSTUAUBUCACDEUDUEUF
+      $.
+
+    $( If in an interchangeability context ` y ` is not free in ` ph ` , the
```

Comment:
Non-blocking: Have you got ( [ x <> y ] ph -> [ y <> x ] ph ) ready at this point?  Then use it to prove ichnf2 from ichnf1.

Reply:
I think ~ichnf2 becomes superfluous then and can be removed.